lib/preboot/require.js

```diff
@@ -1,26 +0,0 @@
-'use strict';
-
-var debug = require('diagnostics')('slay:preboot:require');
-
-/*
- * Lazily require a file or directory as a
- * preboot "middleware".
- */
-module.exports = function (opts) {
-  var fullpath = typeof opts !== 'string' ? opts.fullpath : opts;
-
-  debug('schedule %s', fullpath);
-
-  return function (app, options, callback) {
-    var bootable;
-
-    try {
-      debug('executed %s', fullpath);
-      bootable = require(fullpath);
-    } catch (ex) { // probably we want to throw an error here and halt the application?
-      debug('warning Could not load preboot for %s', fullpath, ex.stack);
-      return callback();
-    }
-    bootable(app, options, callback);
-  };
-};
```

Comment:
Your PR description says that it:

> removes expression-requires where necessary and allows you to pass your preboots in explicitly

Are you intending to *remove* the` require` functionality or simply *augment* it? It may be helpful write some tests to ensure that this PR is not introducing a breaking change.

Reply:
Default behavior will still `require` the same files as before. However, new behavior allows you to pass preboot callbacks in directly so that `slay` doesn’t have to require them via expression.

Default behavior is already tested by the `Running application tests` block.

There was existing behavior to allow paths to be passed in. (That wasn’t tested before but is now.) New behavior is also tested. Those new tests are in the `Preboot sources` block.

Reply:
I had originally removed `builtins.require` but that’s not necessary. Added back in 65391da.